src/components/Identity/customerIds/createCustomerIds.js

```diff
@@ -0,0 +1,88 @@
+import { normalizeCustomerIds, validateCustomerIds } from "../util";
+import {
+  crc32,
+  convertBufferToHex,
+  convertStringToSha256Buffer
+} from "../../../utils";
+import { COOKIE_NAMES } from "../constants";
+import createEvent from "../../DataCollector/createEvent";
+
+const { CUSTOMER_ID_HASH } = COOKIE_NAMES;
+
+export default (cookieJar, lifecycle, network, optIn) => {
+  const updateChecksum = checksum => cookieJar.set(CUSTOMER_ID_HASH, checksum);
+
+  // TODO: FIXME: We shouldn't need an event.
+  const createPayload = event => {
+    const payload = network.createPayload();
+    payload.addEvent(event);
+    return payload;
+  };
+
+  const hash = (originalIds, normalizedIds) => {
+    const idNames = Object.keys(normalizedIds);
+    const idsToHash = idNames.filter(idName => originalIds[idName].hash);
+    const idHashPromises = idsToHash.map(id =>
+      convertStringToSha256Buffer(normalizedIds[id].id)
+    );
+    return Promise.all(idHashPromises).then(hashedIds => {
+      return hashedIds.reduce((finalIds, hashedId, index) => {
+        finalIds[idsToHash[index]].id = convertBufferToHex(hashedId);
+        return finalIds;
+      }, normalizedIds);
+    });
+  };
+
+  const makeServerCall = payload => {
+    return lifecycle.onBeforeDataCollection(payload).then(() => {
+      return network.sendRequest(payload, payload.expectsResponse);
+    });
+  };
+
+  const customerIds = {
+    state: {
```

Comment:
It seems like `state` and `setState` could be encapsulated here (i.e. not returned as part of the public interface for this object)